proto/frequenz/api/common/v1/components.proto

```diff
@@ -75,46 +82,102 @@ enum ComponentCategory {
   COMPONENT_CATEGORY_VOLTAGE_TRANSFORMER = 14;
 }
 
-// Enumerated battery types.
-enum BatteryType {
-  // Unspecified.
-  BATTERY_TYPE_UNSPECIFIED = 0;
+// Metadata specific to a microgrid component.
+message ComponentCategoryMetadataVariant {
+  oneof metadata {
+    frequenz.api.common.v1.components.battery.Battery battery = 1;
+    frequenz.api.common.v1.components.ev_charger.EvCharger ev_charger = 2;
+    frequenz.api.common.v1.components.fuse.Fuse fuse = 3;
+    frequenz.api.common.v1.components.grid.Grid grid = 4;
+    frequenz.api.common.v1.components.inverter.Inverter inverter = 5;
+    frequenz.api.common.v1.components.transformer.VoltageTransformer
+      voltage_transformer = 6;
+  }
+}
+
+// ComponentStatus defines the possible statuses for a component.
+//
+// !!! note
+//     The status indicates the status set by the user via the user interface.
+//     The status is not yet included in the Component messages and should be
+//     added.
+//
+enum ComponentStatus {
+  // The status is unspecified. This should not be used.
+  COMPONENT_STATUS_UNSPECIFIED = 0;
 
-  // Li-ion batteries.
-  BATTERY_TYPE_LI_ION = 1;
+  // The component is active.
+  COMPONENT_STATUS_ACTIVE = 1;
 
-  // Sodium-ion batteries
-  BATTERY_TYPE_NA_ION = 2;
+  // The component is inactive.
+  COMPONENT_STATUS_INACTIVE = 2;
 }
 
-// Enumerated inverter types.
-enum InverterType {
-  // Unspecified.
-  INVERTER_TYPE_UNSPECIFIED = 0;
+// Microgrid electrical component details.
+message Component {
+  // The component ID.
+  uint64 id = 1;
```

Comment:
Forgot to mention earlier: in the original specs, the `id` field has the type `ComponentId`. Since the definition of a component ID was missing, I added the type `uint64`.